### ISSUE TITLE
For devices StreetMeasure does not support, disable according button 

### ISSUE THREAD

**Issue by georg-d** [streetcomplete/StreetComplete#4849]
**Use case**
My device Fairphone 3 is not supported by StreetMeasure. Still, in StreetComplete width/heigth quests, I get the blue camera button which prompts to install StreetMeasure – over and over again. This bothers me.
![20230228 154415-StreetComplete 51 1 measurement quest](https://user-images.githubusercontent.com/45993293/221888112-003eb297-234a-43d5-b055-213dbf5bc297.jpg)

**Proposed Solution**
I would like to have this button vanishing completely or being greyed out – at least for some months, so I might check with newer version of StreetMeasure whether it now supports my device. 

Where could the information come from whether button shall valish/disable? 

1. Obsolete: The according properties file is obviously available in StreetMeasure repository and could be included in StreetComplete. => Not feasible, as the list is closed source.
2. After user got the promt to install StreetMeasure and returned to StreetComplete, SC could check whether SM is now an available app and if not, ask the user whether user wants to be prompted again to install SM – like this, persons how could but do not want to use SM are also not bothered any more.

**Comment by matkoniecz** [comment]
See also #4795

Note that you can disable quests in settings (there are 3 AR ones, 2 about width, 1 about height, all with the measure symbol on icon, you can filter list by asked question).

**Comment by matkoniecz** [comment]
> The according properties file is obviously available in StreetMeasure repository and could be included in StreetComplete.

Not really, for the same reasons as StreetMeasure is a separate app (unless that part is actually open sourced)

**Comment by georg-d** [comment]
@matkoniecz Thanks for linking the related issue; I didn't find it. I did strike through first solution idea.

Here, it's not about disabling the quest, just the installation prompt.

**Comment by matkoniecz** [comment]
> it's not about disabling the quest

but

> For devices StreetMeasure does not support, disable according button

runs into the same issue as disabling the quest run into - problem of reliably checking whether device supports it or not

**Comment by westnordost** [comment]
Duplicate of #4795

**Comment by westnordost** [comment]
> so I might check with newer version of StreetMeasure whether it now supports my device.

It is rather whether the device supports StreetMeasure. Support of certain devices cannot be introduced through some update of that app.

Regarding the suggestion to hide the button under certain circumstances, I do not like it. It's like removing the camera button in the note creation dialog if there is no camera app on the device. I'd find this behavior intransparent.

**Comment by matkoniecz** [comment]
Though prompt to hide this quests, if installation of StreetMeasure failed maybe would be nice?

**Comment by westnordost** [comment]
Yeah

**Comment by matkoniecz** [comment]
Is it possible for SC to reenable/propose reenabling this quest once app is installed? That would mitigate this danger.

Also, wait: is any app able to check whether any other specific app is installed?

**Comment by westnordost** [comment]
> Also, wait: is any app able to check whether any other specific app is installed?

Yes, until Android 10 IIRC. For Android 11+, each app must specify in its manifest for which apps exactly it needs to be able to check whether it is installed.

**Comment by georg-d** [comment]
> So, the above simple check will not suffice.

If app is not present, why not ask users whether they want to be prompted again (see initial post). If measuring app is incompatible or undesired, users can choose "don't ask again", if app is installed in background or in future, user can choose "ask me again". The latter one may be default so users that do not read properly don't loose possibilities.

**Comment by westnordost** [comment]
Well, this would show too early, IMO, as it is a completely normal use case to return to the app while (waiting to) install that app.

**Comment by westnordost** [comment]
I implemented it now like this:

- when the user taps the "measure" button while the streetmeasure app is not installed and then comes back to the app with it still not being installed, a dialog shows that offers to disable the quest

- when the user taps the "measure" button while the streetmeasure app is installed but then the result from the app is either invalid or an error (e.g. if it crashed, as far as I know), a dialog shows that offers to disable the quest

Now, I don't like the first point too much, because some users will have the setting to only install new apps on wifi, but on the other hand, streetmeasure is both very small (2.5MB) and the user can always say no in that dialog.
